### ISSUE TITLE
`@stylistic/unit-case` drops every comment a value or a set of parameters holds

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#233]
Found by the comment oracle written for the plan of 2026-08-20, which fixes every fixture with every rule under every primary option and counts the comments that came out against the comments that went in.

### What minimal example or steps are needed to reproduce the bug?

```css
a { b: 1px /* c */ 2px; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/unit-case": "upper"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

`a { b: 1PX /* c */ 2PX; }` — the units in capitals, and every other character of the declaration left exactly as it was.

### What actually happened?

```css
a { b: 1PX  2PX; }
```

The comment is gone, and nothing is reported. A set of at-rule parameters goes the same way: `@media (min-width: 100px /* c */) { … }` comes back as `@media (min-width: 100PX ) { … }`. Under `postcss-scss` an inline comment is lost the same way, `a { b: 1px // c\n\t2px; }` coming back as `a { b: 1PX \n\t2PX; }`.

### What is the cause?

The rule writes `node.value` and `node.params` directly rather than going through `setDeclarationValue` and `setAtRuleParams`. PostCSS keeps the comments of a value in `raws.value.raw` and hands `decl.value` back cleaned of them, and `postcss-scss` keeps a third copy in `raws.value.scss`, which is the one it prints. Writing the bare property discards both raws, so the value the file gets is the cleaned one.

It is the last rule of the plugin still writing that way. `linebreaks` writes a whole document the same way and loses the same comments, which is filed separately.

**Comment by firefoxic** [comment]
One line of this report is wrong, and the branch that closes it found out: `unit-case` was not the last rule of the plugin writing that way. `number-leading-zero` and `number-no-trailing-zeros` read `decl.value` and `atRule.params` and write the fix back to the same bare properties, and lose the same comments in the same silence. That is [#268](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/268), and it is not closed here.

The corpus of the comment oracle never caught the two because no fixture of it puts a number carrying a zero beside a comment. `a { b: 1px /* c */ 0.50em; }` under `"never"` finds them at once.
